tests/integ/test_marketplace.py

```diff
@@ -186,6 +197,135 @@ def predict_wrapper(endpoint, session):
+        rm=True,
+    )
+    image.tag(ecr_image, tag="latest")
+    _create_repository(ecr_client, algorithm_name)
+
+    # Retry docker image push
+    for _ in retries(3, "Upload docker image to ECR repo", seconds_to_sleep=10):
+        try:
+            docker_client.images.push(
+                ecr_image, auth_config={"username": username, "password": password}
+            )
+            break
+        except requests.exceptions.ConnectionError:
+            # This can happen when we try to create multiple repositories in parallel, so we retry
+            pass
+
+    yield ecr_image
+
+    # Delete repository after the marketplace integration tests complete
+    _delete_repository(ecr_client, algorithm_name)
+
+
+def test_create_model_package(sagemaker_session, boto_session, iris_image):
+
+    # Prepare
+    s3_bucket = sagemaker_session.default_bucket()
+
+    model_name = "my-flower-detection-model"
+    model_description = "This model accepts petal length, petal width, sepal length, sepal width and predicts whether \
+    flower is of type setosa, versicolor, or virginica"
+
+    supported_realtime_inference_instance_types = supported_batch_transform_instance_types = [
+        "ml.m4.xlarge"
+    ]
+    supported_content_types = ["text/csv", "application/json", "application/jsonlines"]
+    supported_response_MIME_types = ["application/json", "text/csv", "application/jsonlines"]
+
+    validation_input_path = "s3://" + s3_bucket + "/validation-input-csv/"
+    validation_output_path = "s3://" + s3_bucket + "/validation-output-csv/"
+
+    iam = boto_session.resource("iam")
+    role = iam.Role("SageMakerRole").arn
+    sm_client = boto_session.client("sagemaker")
+    s3_client = boto_session.client("s3")
+    s3_client.put_object(
+        Bucket=s3_bucket, Key="validation-input-csv/input.csv", Body="5.1, 3.5, 1.4, 0.2"
+    )
+
+    ValidationSpecification = {
+        "ValidationRole": role,
+        "ValidationProfiles": [
+            {
+                "ProfileName": "Validation-test",
+                "TransformJobDefinition": {
+                    "BatchStrategy": "SingleRecord",
+                    "TransformInput": {
+                        "DataSource": {
+                            "S3DataSource": {
+                                "S3DataType": "S3Prefix",
+                                "S3Uri": validation_input_path,
+                            }
+                        },
+                        "ContentType": supported_content_types[0],
+                    },
+                    "TransformOutput": {
+                        "S3OutputPath": validation_output_path,
+                    },
+                    "TransformResources": {
+                        "InstanceType": supported_batch_transform_instance_types[0],
+                        "InstanceCount": 1,
+                    },
+                },
+            },
+        ],
+    }
+
+    # get pre-existing model artifact stored in ECR
```

Comment:
stored in s3?

Reply:
It is stored in ECR created by the iris_image function on line 201